app/views/products/show.html.erb

```diff
@@ -0,0 +1,66 @@
+<% provide :title, @product.title %>
+<div class="clear"></div>
+<section class="main-detail">
+  <div class="container">
+    <div class="row">
+      <div class="pathway">
+        <ul class="path">
+          <li>
+            <%= link_to t(".home"), root_url %>
+          </li>
+          <% if parent_cate(@product.id).present? %>
+            <li><%= link_to parent_cate(@product.id).name, "#" %></li>
+          <% end %>
+          <li><%= link_to category_by_id(@product.id).name, "#" %></li>
```

Comment:
nên check xem method `category_by_id` có bị nil ko chứ chú, chứ nếu find_by bị nil mà `.name` thì null pointer nè

Reply:
e nghĩ là product của e bắt buộc có category rồi nên nếu có product thì chắc chắn có category ạ, còn nếu không có product thì not found rồi ạ

Reply:
>`nên check xem method category_by_id có bị nil ko chứ chú, chứ nếu find_by bị nil mà .name thì null pointer nè`

sao chưa thấy thay đổi gì chú

Reply:
ah, ở đây sao chú ko lấy `@product.category.name` nhỉ, cần gì gọi method thế này